lib/full-site-editing/block-templates.php

```diff
@@ -105,6 +105,35 @@ function _gutenberg_get_template_files( $template_type ) {
 	return $template_files;
 }
 
+/**
+ * Parses wp_template content and injects the current theme's
+ * stylesheet as a theme attribute into each wp_template_part block
+ *
+ * @param string $template_content serialized wp_template content.
+ * @param string $theme the current instances theme stylesheet.
+ *
+ * @return string Updated wp_template content.
+ */
+function _inject_theme_attribute_in_content( $template_content, $theme ) {
+	$new_content = '';
+	$blocks = parse_blocks( $template_content );
+
+	$updated_blocks = array_map(
+		function( $block ) use ( $theme ) {
+			if ('core/template-part' === $block['blockName']) {
+				$block['attrs']['theme'] = $theme;
+			}
+
+			return $block;
+	}, $blocks );
+
+	foreach ( $updated_blocks as $block ) {
+		$new_content = $new_content . serialize_block($block);
+	}
+
+	return $new_content;
```

Comment:
Can we just keep the old content if no block was impacted?

Reply:
Good callout 👍

Reply:
Added a condition to return original template content if no updates are made